src/common/instructions.rs

```diff
@@ -0,0 +1,187 @@
+use anyhow::Result;
+use borsh::BorshDeserialize;
+use solana_client::nonblocking::rpc_client::RpcClient;
+use solana_sdk::pubkey::Pubkey;
+use std::collections::HashMap;
+use std::str::FromStr;
+use flate2::read::ZlibDecoder;
+use std::io::Read;
+
+// Anchor IDL account discriminator
+const ANCHOR_DISCRIMINATOR: &str = "anchor:idl";
+
+use serde::{Deserialize, Serialize};
+
+use super::KoraError;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ProgramInstructionConfig {
+    pub program_id: String,
+    pub instructions: Vec<String>,
+}
+
+#[derive(Debug, Clone, Serialize)]
+pub struct ProgramInstructionConfigWithDiscriminators {
+    pub program_id: String,
+    pub instructions: Vec<String>, // Keep human readable format
+    pub discriminators: HashMap<Vec<u8>, String>, // Map instruction bytes to name/raw disc
+}
+
+pub async fn get_program_instruction_configs_with_discriminators(
+    rpc_client: &RpcClient,
+    program_configs: &[ProgramInstructionConfig],
+) -> Result<Vec<ProgramInstructionConfigWithDiscriminators>, KoraError> {
+    let mut configs_with_discriminators = Vec::new();
+
+    for config in program_configs {
+        let mut discriminator_map = HashMap::new();
+        let mut instruction_names = Vec::new();
+
+        // Process each instruction - could be raw discriminator or name
+        for instruction in &config.instructions {
+            if instruction == "*" {
+                // Wildcard case - allow all instructions
+                instruction_names.push(instruction.clone());
```

Comment:
For wildcard processing, we actually don't need to store any instructions technically, right?

Because later on, we allow any instruction to go through w/o checking discriminator, right?

Reply:
Nothing actionable here, just wanted to flag